plugins/utils/auto-save.nix

```diff
@@ -6,65 +6,94 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.auto-save;
-in
-{
-  options.plugins.auto-save = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "auto-save";
-
-    package = helpers.mkPluginPackageOption "auto-save" pkgs.vimPlugins.auto-save-nvim;
-
-    keymaps = {
-      silent = mkOption {
-        type = types.bool;
-        description = "Whether auto-save keymaps should be silent.";
-        default = false;
-      };
-
-      toggle = helpers.mkNullOrOption types.str "Keymap for running auto-save.";
-    };
-
-    enableAutoSave = helpers.defaultNullOpts.mkBool true ''
-      Whether to start auto-save when the plugin is loaded.
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "auto-save";
+  originalName = "auto-save.nvim";
+  defaultPackage = pkgs.vimPlugins.auto-save-nvim;
+
+  maintainers = [ helpers.maintainers.braindefender ];
+
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "enableAutoSave"
```

Comment:
```suggestion
```

This will need a custom rename warning, since the old name is different to the new name.

You can add this into an `imports = [ ]` next to `optionsRenamedToSettings`:

```nix
imports = let
  basePluginPath = ["plugins" "auto-save"];
  settingsPath = basePluginPath ++ [ "settings" ];
in
  [
    (mkRenamedOptionModule  (basePluginPath ++ [ "enableAutoSave" ]) (settingsPath ++ [ "enabled" ]))
  ];
```

(If you have a few of those, we usually declare a `basePluginPath` and `settingsPath` in a let block, see [this, for example](https://github.com/nix-community/nixvim/blob/4e224d27ae0db889f92ef228ad6cb984b91777fb/plugins/utils/rest.nix#L31-L47))

Reply:
`optionsRenamedToSettings` is [smart enough](https://github.com/nix-community/nixvim/blob/4e224d27ae0db889f92ef228ad6cb984b91777fb/lib/neovim-plugin.nix#L90-L95) to translate camelCase into snake_case, but for anything more complex we need to use [`mkRenamedOptionModule`](https://noogle.dev/f/lib/mkRenamedOptionModule) or [`mkRemovedOptionModule`](https://noogle.dev/f/lib/mkRemovedOptionModule) explicitly.